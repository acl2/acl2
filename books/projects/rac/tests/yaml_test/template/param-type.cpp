// RAC begin

struct A {
  int a;
};
struct B {
  int a;
};

template <int a>
int foo(A b) {
  return a;
}

int bar() {
  B s = {};
  return foo<0>(0);
}

// RAC end
