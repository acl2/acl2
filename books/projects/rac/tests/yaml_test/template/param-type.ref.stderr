param-type.cpp:17 (16-17): Expected `A`, got `int`
 17 |   return foo<0>(0);
    |                 ^
note: Defined here: 
 10 | template <int a>
    | ^^^^^^^^^^^^^^^^^
 11 | int foo(A b) {
    | ^^^^^^^^^^^^
